config/src/main/java/org/axonframework/config/DefaultConfigurer.java

```diff
@@ -540,6 +540,7 @@ public Configuration buildConfiguration() {
         if (!initialized) {
             verifyIdentifierFactory();
             prepareModules();
+            shutdownHandlers.add(new RunnableHandler(0, () -> config.deadlineManager().shutdown()));
```

Comment:
How come this operation doesn't fit in the `prepareModules` method where we add other shutdown handlers?

Reply:
`DeadlineManager` is not a module :)

Reply:
Sounds like we need a `Configurer#configureDeadlineManager` method which will do the job!

Don't think the `buildConfiguration` is the right place to add this shutdown handler to be honest.

Reply:
If you configure a different `DeadlineManager` you'd have to unregister the previous one's `shutdownHandler`. `buildConfiguration` is exactly the point where we can refer to the right (configured single instance) of `DeadlineManager`. It looks a bit "ugly" because `DeadlineManager` is the first non-module component that needs to be shut down when the configuration is shutting down.

Reply:
How wouldn't such a `configureDeadlineManager` be unable to unregister previous shutdown handlers?

Reply:
After some offline discussions and thinking, I think it's justified to keep this as is and provide an adjusted approach when we start with #891. That issue will bring start/shutdown handlers to a far broader audience any how. I'll thus mark this conversation as resolved as a consequence of this decision.